snuba/query/snql/parser.py

```diff
@@ -136,7 +136,7 @@
 
     select_list          = select_columns* (selected_expression)
     select_columns       = selected_expression space* comma
-    selected_expression  = space* low_pri_arithmetic
+    selected_expression  = space* (aliased_tag_column / aliased_subscriptable / aliased_column_name / low_pri_arithmetic)
```

Comment:
Allowing aliases in the group by was intended, since those fields also end up in the results. The function call is more complicated and warrants a second PR because legacy features are relying on function aliases being used in the query itself.